personality-insights/v3.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2016 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var requestFactory = require('../lib/requestwrapper');
+var pick = require('object.pick');
+var extend = require('extend');
+var helper = require('../lib/helper');
+var util = require('util');
+var BaseService = require('../lib/base_service');
+
+/**
+ *
+ * @param options
+ * @constructor
+ */
+function PersonalityInsightsV3(options) {
+  BaseService.call(this, options);
+
+  // Check if 'version_date' was provided
+  if (typeof this._options.version_date === 'undefined') {
+    throw new Error('Argument error: version_date was not specified, use 2016-10-19');
+  }
+  this._options.qs.version = options.version_date;
+
+}
+util.inherits(PersonalityInsightsV3, BaseService);
+PersonalityInsightsV3.prototype.name = 'personality_insights';
+PersonalityInsightsV3.prototype.version = 'v3';
+PersonalityInsightsV3.URL = 'https://gateway.watsonplatform.net/personality-insights/api';
+
+/**
+ * @param params {Object} The parameters to call the service
+ *   The accepted parameters are:
+ *     - text: The text to analyze.
+ *     - content_items: A JSON input (if 'text' not provided).
+ *     - raw_scores: include raw results.
+ *     - consumption_preferences: If true, information about consumption preferences is returned with the results.
+ *     - version_date: The version date. Use 2016-10-19.
```

Comment:
If you document the params like `@param {String} [params.Accept-Language=en] - response language`, then the jsdoc can pull out the details, and editors can offer hints